src/esn.calendar.libs/app/components/event/form/event-form.controller.js

```diff
@@ -156,9 +156,20 @@ function CalEventFormController(
 
       calendarService.listPersonalAndAcceptedDelegationCalendars($scope.calendarHomeId)
         .then(function(calendars) {
+          // Those are the calendars the user can create events within.
           $scope.calendars = calendars;
 
-          return calEventUtils.isNew($scope.editedEvent) ? _.find(calendars, 'selected') : _getCalendarByUniqueId($scope.editedEvent.calendarUniqueId);
+          if (calEventUtils.isNew($scope.editedEvent)) {
+            // This only has a value right after duplicating an event.
+            const eventSourceCalendarId = calEventUtils.getDuplicateEventSource();
+
+            // Check if the event is duplicated and the user owns the source calendar.
+            return eventSourceCalendarId && calendars.filter(({ id }) => id === eventSourceCalendarId) ?
```

Comment:
```calendars.filter(({ id }) => id === eventSourceCalendarId)```

Will this always be true?

Reply:
not if eventSourceCalendarId is a shared calendar.

Reply:
also should be `find` instead of `filter`